Scripts/Runtime/Entities/Lifecycle/EntityLifecycleStatus.cs

```diff
@@ -0,0 +1,251 @@
+        {
+            m_OwningSystem = owningSystem;
+            m_QueryComponentTypes = queryComponentTypes;
+
+            m_Lookup = new AccessControlledValue<NativeParallelHashSet<Entity>>(
+                new NativeParallelHashSet<Entity>(
+                    ChunkUtil.MaxElementsPerChunk<Entity>(),
+                    Allocator.Persistent));
+
+            m_ArrivedEntities = new AccessControlledValue<NativeList<Entity>>(new NativeList<Entity>(ChunkUtil.MaxElementsPerChunk<Entity>(), Allocator.Persistent));
+            m_DepartedEntities = new AccessControlledValue<NativeList<Entity>>(new NativeList<Entity>(ChunkUtil.MaxElementsPerChunk<Entity>(), Allocator.Persistent));
+        }
+
+        protected override void DisposeSelf()
+        {
+            m_Lookup.Dispose();
+            m_ArrivedEntities.Dispose();
+            m_DepartedEntities.Dispose();
+            base.DisposeSelf();
+        }
+
+        public void CreateQuery()
+        {
+            m_Query = m_OwningSystem.GetEntityQuery(m_QueryComponentTypes);
+            m_Query.SetOrderVersionFilter();
+        }
+        
+        //*************************************************************************************************************
+        // PUBLIC API
+        //*************************************************************************************************************
+
+        /// <inheritdoc cref="IEntityLifecycleStatus.AcquireArrivalsAsync"/>
+        public JobHandle AcquireArrivalsAsync(out NativeList<Entity> arrivals)
+        {
+            return m_ArrivedEntities.AcquireAsync(AccessType.SharedRead, out arrivals);
+        }
+
+        /// <inheritdoc cref="IEntityLifecycleStatus.AcquireDeparturesAsync"/>
+        public JobHandle AcquireDeparturesAsync(out NativeList<Entity> departures)
+        {
+            return m_DepartedEntities.AcquireAsync(AccessType.SharedRead, out departures);
+        }
+
+        /// <inheritdoc cref="IEntityLifecycleStatus.ReleaseArrivalsAsync"/>
+        public void ReleaseArrivalsAsync(JobHandle dependsOn)
+        {
+            m_ArrivedEntities.ReleaseAsync(dependsOn);
+        }
+
+        /// <inheritdoc cref="IEntityLifecycleStatus.ReleaseDeparturesAsync"/>
+        public void ReleaseDeparturesAsync(JobHandle dependsOn)
+        {
+            m_DepartedEntities.ReleaseAsync(dependsOn);
+        }
+
+        /// <inheritdoc cref="IEntityLifecycleStatus.AcquireArrivals"/>
+        public AccessControlledValue<NativeList<Entity>>.AccessHandle AcquireArrivals()
+        {
+            return m_ArrivedEntities.AcquireWithHandle(AccessType.SharedRead);
+        }
+
+        /// <inheritdoc cref="IEntityLifecycleStatus.AcquireDepartures"/>
+        public AccessControlledValue<NativeList<Entity>>.AccessHandle AcquireDepartures()
+        {
+            return m_DepartedEntities.AcquireWithHandle(AccessType.SharedRead);
+        }
```

Comment:
Why not just expose the ACVs for `m_ArrivedEntities` and `m_DepartedEntities`?

Reply:
I don't want you to be able to write to them and #195 was made after this so I don't have `IReadOnlyAccessControlledValue` yet. 

Added TODO